### ISSUE TITLE
protobuf.bzl outdated making build with TF_SYSTEM_LIBS=com_google_protobuf fail

### ISSUE THREAD

**Issue by Flamefire** [tensorflow/tensorflow#37864]
**Describe the current behavior**

When using `TF_SYSTEM_LIBS=com_google_protobuf` to configure TensorFlow 2.1 the build fails with 

```
ERROR: /dev/shm/s3248973-EasyBuild/TensorFlow/2.1.0/foss-2019b-Python-3.7.4/TensorFlow/tensorflow-2.1.0/tensorflow/core/profiler/BUILD:51:1: in proto_gen rule //tensorflow/core/profiler:profiler_service_proto_py_genproto: 
Traceback (most recent call last):
        File "/dev/shm/s3248973-EasyBuild/TensorFlow/2.1.0/foss-2019b-Python-3.7.4/TensorFlow/tensorflow-2.1.0/tensorflow/core/profiler/BUILD", line 51
                proto_gen(name = 'profiler_service_proto_py_genproto')
        File "/tmp/easybuild-tmp/eb-NQjKjm/tmp8LJdGE-bazel-build/external/com_google_protobuf/protobuf.bzl", line 110, in _proto_gen_impl
                ctx.actions.run(inputs = inputs, outputs = ctx.out..., <4 more arguments>)
Found tool(s) 'bazel-out/k8-opt/bin/external/grpc/grpc_python_plugin' in inputs. A tool is an input with executable=True set. All tools should be passed using the 'tools' argument instead of 'inputs' in order to make their runfiles availa
ble to the action. This safety check will not be performed once the action is modified to take a 'tools' argument. To temporarily disable this check, set --incompatible_no_support_tools_in_action_inputs=false.
```

Reason is seemingly a very outdated protobuf.bzl which is used with system_libs. It might be enough to update it from the current protobuf repo. Maybe even download that single file only when using system libs

**Comment by Saduf2019** [comment]
@Flamefire 
 please share simple stand alone code before which this error was encountered

**Comment by Flamefire** [comment]
Sure:

- Compile and install protobuf into /usr (or install system package where available)
- `TF_SYSTEM_LIBS=com_google_protobuf ./configure`
- `bazel build //tensorflow/tools/pip_package:build_pip_package`

**Comment by Saduf2019** [comment]
@Flamefire
can you please let us know what OS are you facing this issue in

**Comment by Flamefire** [comment]
Linux Mint 19.1, but this is a Bazel (code) issue, not dependent on the system. The error is

> Found tool(s) 'bazel-out/k8-opt/bin/external/grpc/grpc_python_plugin' in inputs.

The code mentioned is:
https://github.com/tensorflow/tensorflow/blob/6609461732729e9e60567b13d5345bc19fead5d1/third_party/systemlibs/protobuf.bzl#L112-L119

No mention of a "tools" parameter, only "inputs". Upstream code has "tools": https://github.com/protocolbuffers/protobuf/blob/dec4939439d9ca2adf2bb14edccf876c2587faf2/protobuf.bzl#L141-L149

Hence my suggestion to update it in the initial post.

**Comment by Flamefire** [comment]
This seems to be fixed in 2.3.0 by https://github.com/tensorflow/tensorflow/commit/9cee90ac957f2575da2415242c5c127c0dd61430
